Makefile

```diff
@@ -1,22 +1,46 @@
+GOOS ?= `go env GOOS`
+GOPATH ?= `go env GOPATH`
+GOBIN ?= $(GOPATH)/bin
+GO111MODULE = auto
+
 REGISTRY ?= quay.io
 REGISTRY_ACCOUNT ?= kubev2v
 REGISTRY_TAG ?= devel
-CONTROLLER_IMAGE ?= ${REGISTRY}/${REGISTRY_ACCOUNT}/forklift-controller:${REGISTRY_TAG}
-API_IMAGE ?= ${REGISTRY}/${REGISTRY_ACCOUNT}/forklift-api:${REGISTRY_TAG}
-VALIDATION_IMAGE ?= ${REGISTRY}/${REGISTRY_ACCOUNT}/forklift-validation:${REGISTRY_TAG}
-VIRT_V2V_IMAGE ?= ${REGISTRY}/${REGISTRY_ACCOUNT}/forklift-virt-v2v:${REGISTRY_TAG}
-OPERATOR_IMAGE ?= ${REGISTRY}/${REGISTRY_ACCOUNT}/forklift-operator:${REGISTRY_TAG}
-OPERATOR_BUNDLE_IMAGE ?= ${REGISTRY}/${REGISTRY_ACCOUNT}/forklift-operator-bundle:${REGISTRY_TAG}
-OPERATOR_INDEX_IMAGE ?= ${REGISTRY}/${REGISTRY_ACCOUNT}/forklift-operator-index:${REGISTRY_TAG}
-GOOS ?= `go env GOOS`
-GOBIN ?= ${GOPATH}/bin
-GO111MODULE = auto
 
-ifeq (, $(shell which docker))
-    CONTAINER_CMD = podman
-else
-    CONTAINER_CMD = docker
-endif
```

Comment:
this small part is really catching the eye - what do we get out of dropping this? I agree with @nyoxi that it would be better not to require users to install docker or podman-docker but to default to podman

Reply:
I might be wrong but AFAIK `docker` command is currently mandatory for `bazel` to be able to build the images (https://github.com/bazelbuild/rules_docker#setup), so either `docker` or `podman-docker` is mandatory in the environment. My thought was: if `docker` command is going to be required anyway what's the point in having this kind of logic?

On the other hand, we could try  to make `docker_toolchain_configure` in bazel's image builds to point `podman` cmd instead of `docker`, this way we would be able to prefer `podman` and we wouldn't need to install `podman-docker`. Not sure if this will work though.

Reply:
I think I used podman, without docker or podman-docker, after we switched to bazel but I may not remember it correctly (although I also see the path to docker is optional and the statement: `These rules do not require / use Docker for pulling, building, or pushing images` in the link above)

Reply:
In my env the build fails if `docker` command is not available:
```
bazel clean ; make build-operator-index-image
INFO: Starting clean (this may take a while). Consider using --async if the clean takes more than several minutes.
bazel run operator:forklift-operator-index-image \
	--verbose_failures \
	--action_env VERSION=99.0.0 \
	--action_env CHANNELS=development \
	--action_env DEFAULT_CHANNEL=development \
	--action_env OPT_OPTS= \
	--action_env REGISTRY=default-route-openshift-image-registry.apps.ocp4.example.com \
	--action_env REGISTRY_TAG=devel \
	--action_env REGISTRY_ACCOUNT=mmartinv
INFO: Analyzed target //operator:forklift-operator-index-image (104 packages loaded, 9145 targets configured).
INFO: Found 1 target...
Target //operator:forklift-operator-index-image up-to-date:
  bazel-out/k8-fastbuild-ST-4a519fd6d3e4/bin/operator/forklift-operator-index-image-layer.tar
INFO: Elapsed time: 3.277s, Critical Path: 1.92s
INFO: 63 processes: 17 internal, 46 linux-sandbox.
INFO: Build completed successfully, 63 total actions
INFO: Build completed successfully, 63 total actions
error: docker not found; do you need to manually configure the docker toolchain?
make: *** [Makefile:161: build-operator-index-image] Error 1
```

Adding the following to the `WORKSPACE` file:
```
load("@io_bazel_rules_docker//toolchains/docker:toolchain.bzl",
    docker_toolchain_configure="toolchain_configure"
)

docker_toolchain_configure(
  name = "docker_config",
  docker_path="/usr/bin/podman",
  docker_flags = [
    "--log-level=info",
  ],
)
```

seems to fix the issue:
```
bazel clean ; make build-operator-index-image
INFO: Starting clean (this may take a while). Consider using --async if the clean takes more than several minutes.
bazel run operator:forklift-operator-index-image \
	--verbose_failures \
	--action_env VERSION=99.0.0 \
	--action_env CHANNELS=development \
	--action_env DEFAULT_CHANNEL=development \
	--action_env OPT_OPTS= \
	--action_env REGISTRY=default-route-openshift-image-registry.apps.ocp4.example.com \
	--action_env REGISTRY_TAG=devel \
	--action_env REGISTRY_ACCOUNT=mmartinv
INFO: Analyzed target //operator:forklift-operator-index-image (104 packages loaded, 9145 targets configured).
INFO: Found 1 target...
Target //operator:forklift-operator-index-image up-to-date:
  bazel-out/k8-fastbuild-ST-4a519fd6d3e4/bin/operator/forklift-operator-index-image-layer.tar
INFO: Elapsed time: 3.366s, Critical Path: 2.02s
INFO: 63 processes: 17 internal, 46 linux-sandbox.
INFO: Build completed successfully, 63 total actions
INFO: Build completed successfully, 63 total actions
INFO[0000] /usr/bin/podman filtering at log level info  
INFO[0000] Setting parallel job count to 25             
Getting image source signatures
Copying blob 3dddd78e9f35 skipped: already exists  
Copying blob 1baf2ef4355a skipped: already exists  
Copying blob c8ff0b214684 skipped: already exists  
Copying blob 03dea58e7c9a skipped: already exists  
Copying blob cc6e9fe26463 skipped: already exists  
Copying blob 84ff92691f90 skipped: already exists  
Copying blob c456571abc85 skipped: already exists  
Copying config 013957b72a done  
Writing manifest to image destination
Storing signatures
Loaded image: sha256:013957b72a77501557c2925b1053cba6b51c34e94ab3e90a6ef0a94ecc9b0e28
Tagging 013957b72a77501557c2925b1053cba6b51c34e94ab3e90a6ef0a94ecc9b0e28 as bazel/operator:forklift-operator-index-image
INFO[0000] /usr/bin/podman filtering at log level info  
INFO[0000] Setting parallel job count to 25             
```
I also found other problems with `opm render` command using `~/.docker/config.json` by default (https://github.com/operator-framework/operator-registry/issues/919) but I think we could try to configure the bazel toolchain to use podman instead of docker and pass `DOCKER_CONFIG` env var to `opm` depending on which runtime is in use.

Reply:
ack, but then we'll need to install podman on the github-hosted runners?